lib/zomato/client.rb

```diff
@@ -0,0 +1,57 @@
+module Zomato
+  class Client
+    NEW_YORK_CITY_ID = 280
+
+    attr_reader :restaurants
+
+    def initialize(options = {})
+      @restaurants = []
+      @city_id = options[:city_id] || NEW_YORK_CITY_ID
+      @api_key = options[:api_key] || ENV['ZOMATO_API_KEY']
+    end
+
+    def search_restaurants
+      return restaurants if restaurants.present?
+
+      [0, 20, 40, 60, 80].each do |start_offset|
+        raw_restaurants = fetch(start_offset: start_offset)
+        transform_and_populate(raw_restaurants)
+      end
+
+      restaurants
+    end
+
+    private
+
+    def fetch(start_offset: 0)
+      response = RestClient::Request.execute(
+        method: :get,
+        url: 'https://developers.zomato.com/api/v2.1/search',
```

Comment:
minor: I would put the url in an env file key rather than use it hard coded like so